lib/src/component/with_transition.dart

```diff
@@ -0,0 +1,410 @@
+// Copyright 2020 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library over_react.abstract_controlled_transition;
+
+import 'dart:async';
+import 'dart:html';
+
+import 'package:over_react/over_react.dart';
+import 'package:over_react/components.dart' as v2;
+
+part 'with_transition.over_react.g.dart';
+
+mixin WithTransitionPropsMixin on UiProps {
+  /// Controls the visibility of the component.
+  ///
+  /// * When set to `true` - the component will begin to transition into a visible state.
+  /// * When set to `false` - the component will begin to transition into a hidden state.
+  ///
+  /// > Default: `false`
+  ///
+  /// > See: [WithTransition] for example usage.
+  bool isShown;
+
+  /// An optional map of props to apply to the single child based on the current transition phase.
+  ///
+  /// This provides you with the ability to add CSS classes / other DOM attributes based on the granular
+  /// transition states that are confined within the [WithTransition] component without having to
+  /// utilize state change callbacks to update your component's state.
+  ///
+  /// __NOTE:__ These values will override any prop values set on the child itself if the
+  /// key(s) in the `Map` are the same.
+  ///
+  /// __Example:__
+  ///
+  /// ```dart
+  /// UiFactory<WithTransitionExampleProps> WithTransitionExample = uiFunction(
+  ///   (props) {
+  ///     final isShown = useState(false);
+  ///
+  ///     (WithTransition()
```

Comment:
#nit this expression is unused;
```suggestion
  ///     return (WithTransition()
```

Reply:
Addressed in https://github.com/Workiva/over_react/pull/621/commits/967d93dd4c6cbb6fac367b02d128a6002613d8b8